src/Pages/Competition.jsx

```diff
@@ -0,0 +1,28 @@
+import { useParams } from "react-router-dom"
+import { useEffect } from "react"
+
+function Competition({setCompetitions}) {
+    const { id } = useParams();
+
+    function getCompetition() {
+        const url = `http://localhost:3030/competition/${id}`;
+        fetch(url)
+          .then((res) => res.json())
+          .then((Data) => {
+            console.log(Data)
+            setCompetitions(Data.data)
```

Comment:
Nice. Be aware that this fetch is getting a single competition, and then you're calling `setCompetitions` which will transform the original array of competitions into a single object (a competition).

I don't think you need `setCompetitions` here. You might instead have a
`const [competition, setCompetition] = useState({});` in this Competition component, and call `setCompetition` in the fetch.then instead

Reply:
Will correct that now